internal/helm/helm.go

```diff
@@ -197,6 +203,33 @@ func newChartFromDirectory(directory string) (*chart.Chart, error) {
 	return loader.LoadDir(directory)
 }
 
+func newChartFromRemote(ciliumVersion semver2.Version) (*chart.Chart, error) {
+	pull := action.NewPullWithOpts(action.WithConfig(new(action.Configuration)))
+	pull.Settings = settings
+	pull.RepoURL = ciliumChart
+	pull.Version = ciliumVersion.String()
+
+	tempDir, err := os.MkdirTemp("", "cilium-cli")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
from the [`os.MkdirTemp`](https://pkg.go.dev/os#MkdirTemp) doc:

> […] It is the caller's responsibility to remove the directory when it is no longer needed.

```suggestion
	}
	defer os.RemoveAll(tempDir)
```